File: benchmarks/ruby/primes.rb
LIMIT = 100_000

sieve = Array.new(LIMIT + 1, true)
sieve[0] = false
sieve[1] = false

p = 2
while p * p <= LIMIT
  if sieve[p]
    m = p * p
    while m <= LIMIT
      sieve[m] = false
      m += p
    end
  end
  p += 1
end

count = 0
(2..LIMIT).each do |k|
  count += 1 if sieve[k]
end

puts count

File: benchmarks/lua/primes.lua
local LIMIT = 100000

local sieve = {}
for i = 0, LIMIT do
  sieve[i] = true
end
sieve[0] = false
sieve[1] = false

local p = 2
while p * p <= LIMIT do
  if sieve[p] then
    local m = p * p
    while m <= LIMIT do
      sieve[m] = false
      m = m + p
    end
  end
  p = p + 1
end

local count = 0
for k = 2, LIMIT do
  if sieve[k] then
    count = count + 1
  end
end

print(count)
